### PR TITLE
Windows build doc update

### PR DESCRIPTION
**Pull request by sipsorcery**
Attempted to make introduction clearer. Added note about required location of source code.

The note about the source code location should stop the issues being logged about the automake errors when building the dependencies.

### CODE REVIEW

**Comment by laanwj** [comment]
LGTM, utACK https://github.com/bitcoin/bitcoin/pull/11704/commits/7605da94501e40592d626f36f42dd896597fff9b

File: doc/build-windows.md [hunk 1]
```diff
@@ -94,6 +98,18 @@ Ubuntu Zesty 17.04 <sup>[2](#footnote2)</sup>:
 
 Once the tool chain is installed the build steps are common:
 
+Note that for WSL the Bitcoin Core source path MUST be somewhere in the default mount file system, for
```
Comment (Thoragh) [inline doc/build-windows.md]: This should also be written under building for 32-bit windows.

Comment (sipsorcery) [reply doc/build-windows.md]: Done.

**Comment by fanquake** [comment]
Please update the commit message to something more meaningful. Include [Doc] and mention Windows etc.

File: doc/build-windows.md [hunk 2]
```diff
@@ -3,29 +3,28 @@ WINDOWS BUILD NOTES
...
-
-Building on Ubuntu Trusty 14.04 is recommended.
-At the time of writing the Windows Subsystem for Linux installs Ubuntu Xenial 16.04. The default cross
-compiler package for Ubuntu Xenial does not produce working executables for some of the bitcoin applications.
-It is possible to build on Ubuntu Xenial by installing the cross compiler packages from
-Ubuntu Zesty, see the steps below.
-Building on Ubuntu Zesty 17.04 up to 17.10 has been verified to work.
+The options known to work for building Bitcoin Core on Windows are:
+
+* On Linux using the (Mingw-w64)[https://mingw-w64.org/doku.php] cross compiler tool chain. Ubuntu Trusty 14.04 is recommended
```
Comment (fanquake) [inline doc/build-windows.md]: Need to swap parens and square brackets on this link

File: doc/build-windows.md [hunk 3]
```diff
@@ -3,29 +3,28 @@ WINDOWS BUILD NOTES
...
+Subsystem for Linux (WSL)](https://msdn.microsoft.com/commandline/wsl/about) and the Mingw-w64 cross compiler tool chain.
+
+Other options which may work but which have not been extensively tested are (please contribute instructions):
 
-While there are potentially a number of ways to build on Windows (for example using msys / mingw-w64),
-using the Windows Subsystem For Linux is the most straightforward. If you are building with
-another method, please contribute the instructions here for others who are running versions
-of Windows that are not compatible with the Windows Subsystem for Linux.
+* On Windows using a POSIX compatibility layer application such as [cygwin](http://www.cygwin.com/) or [msys2](http://www.msys2.org/).
+* On Windows using a native compiler tool chain such as (Visual Studio)[https://www.visualstudio.com].
```
Comment (fanquake) [inline doc/build-windows.md]: same here

**Comment by laanwj** [comment]
Thanks, merged via ea68190

**Pull request closed without merging**